src/vm/docs/contracts.rs

```diff
@@ -0,0 +1,238 @@
+use chainstate::stacks::boot::STACKS_BOOT_CODE_MAINNET;
+use vm::analysis::{mem_type_check, ContractAnalysis};
+use vm::docs::{get_input_type_string, get_output_type_string, get_signature};
+use vm::types::{FunctionType, Value};
+
+use vm::execute;
+
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::iter::FromIterator;
+
+#[derive(Serialize)]
+struct ContractRef {
+    public_functions: Vec<FunctionRef>,
+    read_only_functions: Vec<FunctionRef>,
+    error_codes: Vec<ErrorCode>,
+}
+
+#[derive(Serialize)]
+struct FunctionRef {
+    name: String,
+    input_type: String,
+    output_type: String,
+    signature: String,
+    description: String,
+}
+
+#[derive(Serialize)]
+struct ErrorCode {
+    name: String,
+    #[serde(rename = "type")]
+    value_type: String,
+    value: String,
+}
+
+struct ContractSupportDocs {
+    descriptions: HashMap<&'static str, &'static str>,
+    skip_func_display: HashSet<&'static str>,
+}
+
+fn make_contract_support_docs() -> HashMap<&'static str, ContractSupportDocs> {
```

Comment:
This is fine for now, but do we want to extract the comments directly from the PoX contract?  Maybe there could be a docstring annotation -- e.g. `;;;` indicates a docstring in Clarity?

Reply:
Yeah, that would be a nice feature, but would require some reworking of the parser (it currently doesn't parse comments at all). Will open an issue for that.